libs/common-hooks/src/useDismissableBanner.ts

```diff
@@ -0,0 +1,29 @@
+import { useAtom } from 'jotai'
+import { atomWithStorage } from 'jotai/utils'
+import { useCallback } from 'react'
+
+import { getJotaiMergerStorage } from '@cowprotocol/core'
+
+type ClosableBannersState = Record<string, true | undefined>
+
+const DEFAULT_STATE: ClosableBannersState = {}
+
+const closableBannersStateAtom = atomWithStorage<ClosableBannersState>(
+  'closableBanners:v0',
+  DEFAULT_STATE,
+  getJotaiMergerStorage()
+)
+
+export function useDismissableBanner(bannerId: string | undefined) {
```

Comment:
We already have it in `libs/ui/src/pure/ClosableBanner/index.tsx`

Reply:
Aaah, I see what are you doing.
So, I suggest creating a new components in `lib/ui`:

```
function DismissableInlineBanner() {
    ...
    return ClosableBanner(key, (onClose) => {
        return <InlineBanner onClose="onClose" />
    })
}
```

Composition is preferred over additional parameters.